File: UI/src/main/java/ca/warp7/rt/java/boards/BoardsFeature.java
package ca.warp7.rt.java.boards;

import ca.warp7.rt.core.ft.Feature;
import ca.warp7.rt.core.ft.FeatureItemTab;
import javafx.scene.Parent;

import java.util.Collections;
import java.util.List;

public class BoardsFeature implements Feature {

    @Override
    public List<FeatureItemTab> getLoadedTabs() {
        return Collections.singletonList(new FeatureItemTab(
                "Scouting Boards", "fas-clipboard:18:gray", getFeatureId(),
                FeatureItemTab.Group.SingleTab, ""
        ));
    }

    @Override
    public String getFeatureId() {
        return "boards";
    }

    @Override
    public Parent onOpenTab(FeatureItemTab tab) {
        return null;
    }

    @Override
    public boolean onCloseRequest() {
        return true;
    }
}

File: UI/src/main/kotlin/ca/warp7/rt/ext/boards/BoardsFeature.kt
package ca.warp7.rt.ext.boards

import ca.warp7.rt.core.feature.Feature
import ca.warp7.rt.core.feature.FeatureItemTab
import ca.warp7.rt.core.feature.FeatureLink

class BoardsFeature : Feature {
    override val link = FeatureLink("Scouting Boards", "fas-clipboard", 18)

    override val loadedTabs
        get() = listOf(FeatureItemTab(
                "Scouting Boards", "fas-clipboard:18:gray", this.featureId,
                FeatureItemTab.Group.SingleTab, ""
        ))

    override val featureId get() = "boards"
}
